# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/xhr/XMLHttpRequestMainThread.cpp
# Commit: e2e4557b8d3d
# Full Hash: e2e4557b8d3d2d82d246a1333012340a78e37259
# Author: Joshua Marshall <jmarshall@mozilla.com>
# Date: 2024-02-28 04:30:28
# Description:
#   Bug 1877921 - Own WorkerRef during DebugWorkerRefs r=dom-worker-reviewers,asuth
#   
#   The Proxy is released on the Worker thread when an iframe with a Worker is destroyed.  This removes the ThreadSafeWorkerRef from the XMLHttpRequestMainThread.  Then the main thread destroys the docshell, calling XMLHttpRequestMainThread::Abort, which relies on the Worker still being alive.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202454
# ==============================================================================

diff -r 4e63443a1754 -r e2e4557b8d3d dom/xhr/XMLHttpRequestMainThread.cpp
--- a/dom/xhr/XMLHttpRequestMainThread.cpp	Tue Feb 27 15:36:28 2024 +0000
+++ b/dom/xhr/XMLHttpRequestMainThread.cpp	Tue Feb 27 15:53:15 2024 +0000
@@ -185,7 +185,7 @@
 // invoked for increased scrutability.  Save the previous value on the stack.
 namespace {
 struct DebugWorkerRefs {
-  RefPtr<ThreadSafeWorkerRef>& mTSWorkerRef;
+  RefPtr<ThreadSafeWorkerRef> mTSWorkerRef;
   nsCString mPrev;
 
   DebugWorkerRefs(RefPtr<ThreadSafeWorkerRef>& aTSWorkerRef,
